driver/src/main.rs

```diff
@@ -102,6 +102,10 @@ struct Options {
     #[structopt(long, env = "ORDERBOOK_FILTER", default_value = "{}")]
     orderbook_filter: OrderbookFilter,
 
+    /// Primary method for orderbook retrieval ("Paginated" or "OnchainFiltered")
+    #[structopt(long, env = "PRIMARY_ORDERBOOK", default_value = "Paginated")]
+    primary_orderbook: String,
```

Comment:
Can you make this typesafe by doing something similar to what we have with `SOLVER_TYPE`. Basically creating an enum that implements FromStr. This way the logic of what cases are possible is colocated with the orderbook module and not inlined in this large main file.